[CI] SharedBlobCacheWarmingServiceTests testIdLookupPreWarming failing

**Build Scans:**
- [elasticsearch-intake #39405 / part2](https://gradle-enterprise.elastic.co/s/3idcr4ynmkkli)
- [elasticsearch-pull-request #143512 / part-2](https://gradle-enterprise.elastic.co/s/yq5dcpyouapig)
- [elasticsearch-pull-request #143512 / part-2](https://gradle-enterprise.elastic.co/s/3fu3w5bv7ykgc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:test" --tests "org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceTests.testIdLookupPreWarming" -Dtests.seed=58FCC699F6468A3E -Dtests.locale=ks-Deva -Dtests.timezone=Africa/Algiers -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testIdLookupPreWarming'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: _0_0.tim region 7 (after header) should NOT be cached
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testIdLookupPreWarming
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test testIdLookupPreWarming (0.4% fail rate in 796 executions)
- [main] 2 failures in step part-2 (0.4% fail rate in 464 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -557,9 +557,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.MlDailyMaintenanceServiceIT
   method: testTriggerCloseIdleJobsWithStoppedDatafeeds
   issue: https://github.com/elastic/elasticsearch/issues/148758
-- class: org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceTests
-  method: testIdLookupPreWarming
-  issue: https://github.com/elastic/elasticsearch/issues/148760
 - class: org.elasticsearch.xpack.stateless.recovery.IndexingShardRelocationIT
   method: testRelocateIndexingShardWithActionFailures
   issue: https://github.com/elastic/elasticsearch/issues/148782

--- a/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceTests.java
+++ b/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceTests.java
@@ -1282,9 +1282,7 @@ public class SharedBlobCacheWarmingServiceTests extends ESTestCase {
                     // when the 16-byte Lucene footer straddles a region boundary).
                     int firstUncachedRegion = lastHeaderRegion + 1;
                     // footer occupies [fileLength - footerLength, fileLength); use its start region as the guard
-                    int footerStartRegion = node.sharedCacheService.getRegion(
-                        loc.offset() + loc.fileLength() - CodecUtil.footerLength()
-                    );
+                    int footerStartRegion = node.sharedCacheService.getRegion(loc.offset() + loc.fileLength() - CodecUtil.footerLength());
                     if (firstUncachedRegion < footerStartRegion) {
                         long uncachedRegionStart = (long) firstUncachedRegion * regionSize;
                         assertFalse(

--- a/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceTests.java
+++ b/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceTests.java
@@ -7,6 +7,7 @@
 
 package org.elasticsearch.xpack.stateless.cache;
 
+import org.apache.lucene.codecs.CodecUtil;
 import org.apache.lucene.store.IOContext;
 import org.apache.lucene.util.SetOnce;
 import org.elasticsearch.action.ActionListener;
@@ -1276,10 +1277,15 @@ public class SharedBlobCacheWarmingServiceTests extends ESTestCase {
                             cacheFile.tryRead(ByteBuffer.allocate(Math.toIntExact(readEnd - readStart)), readStart)
                         );
                     }
-                    // The region immediately after the header should NOT be cached, but we must skip the footer region.
+                    // The region immediately after the header should NOT be cached, but we must skip
+                    // any region covered by the footer (which may start one region before the file's last region
+                    // when the 16-byte Lucene footer straddles a region boundary).
                     int firstUncachedRegion = lastHeaderRegion + 1;
-                    int footerRegion = node.sharedCacheService.getEndingRegion(loc.offset() + loc.fileLength());
-                    if (firstUncachedRegion < footerRegion) {
+                    // footer occupies [fileLength - footerLength, fileLength); use its start region as the guard
+                    int footerStartRegion = node.sharedCacheService.getRegion(
+                        loc.offset() + loc.fileLength() - CodecUtil.footerLength()
+                    );
+                    if (firstUncachedRegion < footerStartRegion) {
                         long uncachedRegionStart = (long) firstUncachedRegion * regionSize;
                         assertFalse(
                             fileName + " region " + firstUncachedRegion + " (after header) should NOT be cached",

